.Lib9c.Tools/SubCommand/State.cs

```diff
@@ -206,132 +205,66 @@ public void Dispose()
                 Directory.CreateDirectory(_dumpPath);
                 _messageWriter.WriteLine("Total keys: {0}", _dictionary.Count);
                 long i = 0;
-                foreach (KeyValuePair<byte[], byte[]> pair in _dictionary)
+                foreach (KeyValuePair<KeyBytes, byte[]> pair in _dictionary)
                 {
                     string subDir = Path.Join(
                         _dumpPath,
-                        pair.Key.LongLength.ToString(),
-                        pair.Key.Length > 0 ? $"{pair.Key[0]:x2}" : "_"
+                        pair.Key.Length.ToString(),
+                        pair.Key.Length > 0 ? $"{pair.Key.ToByteArray()[0]:x2}" : "_"
```

Comment:
`KeyBytes.ToByteArray()` method internally makes a copy of its internal buffer whereas `KeyBytes.ByteArray` property simply returns its internal buffer (and that's why one is a method and other one is a property).

```suggestion
                        pair.Key.Length > 0 ? $"{pair.Key.ByteArray[0]:x2}" : "_"
```